File: src/Z.Compression/GZip/System.IO.FileInfo/FileInfo.CreateGZip.cs
// Description: C# Extension Methods Library to enhances the .NET Framework by adding hundreds of new methods. It drastically increases developers productivity and code readability. Support C# and VB.NET
// Website & Documentation: https://github.com/zzzprojects/Z.ExtensionMethods
// Forum: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License: https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: http://www.zzzprojects.com/
// Copyright © ZZZ Projects Inc. 2014 - 2016. All rights reserved.
using System.IO;
using System.IO.Compression;

public static partial class Extensions
{
    /// <summary>
    ///     A FileInfo extension method that creates a zip file.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    public static void CreateGZip(this FileInfo @this)
    {
        using (FileStream originalFileStream = @this.OpenRead())
        {
            using (FileStream compressedFileStream = File.Create(@this.FullName + ".gz"))
            {
                using (var compressionStream = new GZipStream(compressedFileStream, CompressionMode.Compress))
                {
                    originalFileStream.CopyTo(compressionStream);
                }
            }
        }
    }

    /// <summary>
    ///     A FileInfo extension method that creates a zip file.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <param name="destination">Destination for the zip.</param>
    public static void CreateGZip(this FileInfo @this, string destination)
    {
        using (FileStream originalFileStream = @this.OpenRead())
        {
            using (FileStream compressedFileStream = File.Create(destination))
            {
                using (var compressionStream = new GZipStream(compressedFileStream, CompressionMode.Compress))
                {
                    originalFileStream.CopyTo(compressionStream);
                }
            }
        }
    }

    /// <summary>
    ///     A FileInfo extension method that creates a zip file.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <param name="destination">Destination for the zip.</param>
    public static void CreateGZip(this FileInfo @this, FileInfo destination)
    {
        using (FileStream originalFileStream = @this.OpenRead())
        {
            using (FileStream compressedFileStream = File.Create(destination.FullName))
            {
                using (var compressionStream = new GZipStream(compressedFileStream, CompressionMode.Compress))
                {
                    originalFileStream.CopyTo(compressionStream);
                }
            }
        }
    }
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Compression/GZip/System.IO.FileInfo/FileInfo.CreateGZip.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Imports System.IO
Imports System.IO.Compression

Public Module Extensions_39

	''' <summary>
	'''     A FileInfo extension method that creates a zip file.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	<System.Runtime.CompilerServices.Extension> _
	Public Sub CreateGZip(this As FileInfo)
		Using originalFileStream As FileStream = this.OpenRead()
			Using compressedFileStream As FileStream = File.Create(this.FullName + ".gz")
				Using compressionStream = New GZipStream(compressedFileStream, CompressionMode.Compress)
					originalFileStream.CopyTo(compressionStream)
				End Using
			End Using
		End Using
	End Sub

	''' <summary>
	'''     A FileInfo extension method that creates a zip file.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <param name="destination">Destination for the zip.</param>
	<System.Runtime.CompilerServices.Extension> _
	Public Sub CreateGZip(this As FileInfo, destination As String)
		Using originalFileStream As FileStream = this.OpenRead()
			Using compressedFileStream As FileStream = File.Create(destination)
				Using compressionStream = New GZipStream(compressedFileStream, CompressionMode.Compress)
					originalFileStream.CopyTo(compressionStream)
				End Using
			End Using
		End Using
	End Sub

	''' <summary>
	'''     A FileInfo extension method that creates a zip file.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <param name="destination">Destination for the zip.</param>
	<System.Runtime.CompilerServices.Extension> _
	Public Sub CreateGZip(this As FileInfo, destination As FileInfo)
		Using originalFileStream As FileStream = this.OpenRead()
			Using compressedFileStream As FileStream = File.Create(destination.FullName)
				Using compressionStream = New GZipStream(compressedFileStream, CompressionMode.Compress)
					originalFileStream.CopyTo(compressionStream)
				End Using
			End Using
		End Using
	End Sub
End Module


